CONTRIBUTING.md

```diff
@@ -49,15 +49,88 @@ If your pull request contains any of the below, it's less likely to be merged:
+    - [`IUpdater<ConfigFile>`](src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/IUpdater.cs) is covered in more depth in [extensibility docs](docs/extensibility.md#updaters).
+  - [`RazorUpdaterStep`](src/extensions/web/Microsoft.DotNet.UpgradeAssistant.Steps.Razor/RazorUpdaterStep.cs) updates Razor views (cshtml files) in the project using registered `IUpdater<RazorCodeDocument>` implementations.
+    - [`IUpdater<RazorCodeDocument>`](src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/IUpdater.cs) is covered in more depth in [extensibility docs](docs/extensibility.md#updaters).
+  - [`NextProjectStep`](src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Solution/NextProjectStep.cs) runs after upgrading a specific projects is complete and clears the current project from the upgrade context. This will cause most project-specific steps to reset their state the next time the UpgraderManager checks their applicability, causing them to run again on the next project the user selects.
+  - [`FinalizeSolutionStep`](src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Solution/FinalizeSolutionStep.cs) is intended to run as the last step of the upgrade process and performs cleanup activities such as deleting the .upgrade-assistant state file.
+
+### Other services
+
+- [`IUpgradeReadyCheck`](src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/IUpgradeReadyCheck.cs) implementations are used by the `CurrentProjectSelectionStep` to validate that projects are able to be meaningfully upgraded by Upgrade Assistant and can be successfully loaded.
+- [Roslyn analyzers and code fix providers](src/extensions/default/analyzers) are used by the `SourceUpdaterStep` and `RazorUpdaterStep` to identify patterns in source code that users will need to update and, if a code fix provider is available, will automatically make those fixes. Extensions can add their own analyzers and code fix providers to extend this functionality.
+
+### TFM selection
+
+- [`ITargetFrameworkSelector`](src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/ITargetFrameworkSelector.cs) implementations determine the recommended TFM for given input projects. The default implementation ([`TargetFrameworkSelector`](src/components/Microsoft.DotNet.UpgradeAssistant/TargetFramework/TargetFrameworkSelector.cs)) chooses .NET Standard 2.0 whenever possible (since it is the most portable target for modern development) and uses a series of [`ITargetFrameworkSelectorFilter`](src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/ITargetFrameworkSelectorFilter.cs) implementations to update this baseline until the correct TFM is found. These filters (which can be easily added to by registering additional filters in DI) look at factors like project output type, dependencies' TFMs, and project properties. `ITargetFrameworkSelectorFilter` implementations in the default extensions include:
```

Comment:
Having this the same level as the filters feels a little weird. Maybe structuring sections as `Abstractions` or `Services` and `Example implementations`

Reply:
I have the filter implementations as sub-bullets. Is that what you mean?